CMSgov/mint-app
# [NOREF] - Only submit dirty fields when sending mutations on forms

NOREF

## Changes and Description

- Updated all forms to only pass dirty fields to update mutations

Currently, for IT Solutions, needs questions were being marked as 'Answered' because each form was sending the entire payload even if a field hadn't been touched.  Not we can have certainty over fields that have truly been answered.

## Notes

Formik does indeed have  a `dirty` value, but not per input - only per form.  This PR created a util function to return only dirty fields/values to mutation

## How to test this change

- Submit forms for question pertaining to IT Solutions without actually answering the question
- Verify on IT Solutions, that the relevant need still says `Not Answered`

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
